src/xml.rs

```diff
@@ -1611,6 +1684,98 @@ impl DataArray {
         })
     }
 
+    pub fn parse_face_offset_data(
+        self,
+        _appended: Option<&AppendedData>,
+        ei: EncodingInfo,
+    ) -> std::result::Result<model::FieldArray, ValidationError> {
+        let DataArray {
+            name,
+            scalar_type,
+            format,
+            num_comp,
+            data,
+            ..
+        } = self;
+        let final_data;
+        match format {
+            DataArrayFormat::Appended => {
+                return Err(ValidationError::Unsupported);
+            }
+            DataArrayFormat::Binary => {
+                let header_bytes = ei.header_type.size();
+                use model::IOBuffer;
+                if matches!(ei.compressor, Compressor::None) {
+                    // First byte gives the bytes
+                    let bytes =
+                        BASE64_STANDARD.decode(data.into_iter().next().unwrap().into_string())?;
```

Comment:
Maybe change this `unwrap` to a more descriptive `expect`

Reply:
As well as the one on line 1757

Reply:
Here I think the `into_string` that returns a string should be changed to a `as_str` method that returns a `&str`. Then you can do `data[0].as_str()` which is much more idomatic